File: Seori/LeetCode 75/Array_String/151. Reverse Words in a String.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # 공백을 기준으로 split하고, 빈 값을 제거하여 다시 역순으로 join한다.
        list_s = s.split(" ")
        while "" in list_s:
            list_s.remove("")
        return " ".join(list_s[::-1])

File: Lucy/2024/LeetCode 75/Array_String/151_Reverse_Words_in_a_String.ts
/**
 * 151. Reverse Words in a String
 *
 * url: https://leetcode.com/problems/reverse-words-in-a-string/?envType=study-plan-v2&envId=leetcode-75
 * difficulty: Medium
 * topic:
 * date: 2024.10.16(WED)
 */

function reverseWords(s: string): string {
  const copied = s
    .trim()
    .split(" ")
    .filter((s) => s.length > 0);

  return copied.reverse().join(" ");
}

console.log(reverseWords("the sky is blue")); // "blue is sky the"
console.log(reverseWords("  hello world  ")); // "world hello"
console.log(reverseWords("a good   example")); // "example good a"
